x-pack/plugins/enterprise_search/public/applications/app_search/components/crawler/components/crawl_custom_settings_flyout/crawl_custom_settings_flyout_logic.ts

```diff
@@ -0,0 +1,224 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { kea, MakeLogicType } from 'kea';
+
+import { flashAPIErrors } from '../../../../../shared/flash_messages';
+import { HttpLogic } from '../../../../../shared/http';
+import { EngineLogic } from '../../../engine';
+
+import { CrawlerLogic } from '../../crawler_logic';
+import { DomainConfig, DomainConfigFromServer } from '../../types';
+import { domainConfigServerToClient } from '../../utils';
+import { extractDomainAndEntryPointFromUrl } from '../add_domain/utils';
+
+export interface CrawlCustomSettingsFlyoutLogicValues {
+  customEntryPointUrls: string[];
+  customSitemapUrls: string[];
+  domainUrls: string[];
+  domainConfigs: DomainConfig[];
+  domainConfigMap: {
+    [key: string]: DomainConfig;
+  };
+  entryPointUrls: string[];
+  includeRobotsTxt: boolean;
+  isDataLoading: boolean;
+  isFormSubmitting: boolean;
+  isFlyoutVisible: boolean;
+  maxCrawlDepth: number;
+  selectedDomainUrls: string[];
+  selectedEntryPointUrls: string[];
+  selectedSitemapUrls: string[];
+  sitemapUrls: string[];
+}
+
+export interface CrawlCustomSettingsFlyoutLogicActions {
+  fetchDomainConfigData(): void;
+  hideFlyout(): void;
+  onRecieveDomainConfigData(domainConfigs: DomainConfig[]): { domainConfigs: DomainConfig[] };
+  onSelectCustomEntryPointUrls(entryPointUrls: string[]): { entryPointUrls: string[] };
+  onSelectCustomSitemapUrls(sitemapUrls: string[]): { sitemapUrls: string[] };
+  onSelectDomainUrls(domainUrls: string[]): { domainUrls: string[] };
+  onSelectEntryPointUrls(entryPointUrls: string[]): { entryPointUrls: string[] };
+  onSelectMaxCrawlDepth(maxCrawlDepth: number): { maxCrawlDepth: number };
+  onSelectSitemapUrls(sitemapUrls: string[]): { sitemapUrls: string[] };
+  showFlyout(): void;
+  startCustomCrawl(): void;
+  toggleIncludeRobotsTxt(): void;
+}
+
+const filterSeedUrlsByDomainUrls = (seedUrls: string[], domainUrls: string[]): string[] => {
+  const domainUrlMap = domainUrls.reduce(
+    (acc, domainUrl) => ({ ...acc, [domainUrl]: true }),
+    {} as { [key: string]: boolean }
+  );
+
+  return seedUrls.filter((seedUrl) => {
+    const { domain } = extractDomainAndEntryPointFromUrl(seedUrl);
```

Comment:
I'm wondering if it would be more robust to filter the response from the `/domain_configs` endpoints based on the domain, instead of trying to match the domain names?

Reply:
I need to think about how to do this in the context of kea logic a little more.  There's probably a way to accomplish this without having to pass `domainConfigs` in through the `onSelectDomainUrls` action (using a selector?) but its not totally clear to me at the moment.